python/plugins/processing/algs/gdal/ClipRasterByExtent.py

```diff
@@ -180,21 +183,55 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
                     parameters[self.INPUT] if self.INPUT in parameters else "INPUT"
                 )
             )
-        input_details = GdalUtils.gdal_connection_details_from_layer(inLayer)
 
         bbox = self.parameterAsExtent(parameters, self.EXTENT, context, inLayer.crs())
-        override_crs = self.parameterAsBoolean(parameters, self.OVERCRS, context)
-        if self.NODATA in parameters and parameters[self.NODATA] is not None:
-            nodata = self.parameterAsDouble(parameters, self.NODATA, context)
-        else:
-            nodata = None
-        options = self.parameterAsString(parameters, self.OPTIONS, context)
-        out = self.parameterAsOutputLayer(parameters, self.OUTPUT, context)
-        self.setOutputValue(self.OUTPUT, out)
+        if bbox.isNull() or bbox.isEmpty():
+            raise QgsProcessingException(
+                "Invalid extent {}".format(
+                    parameters[self.EXTENT] if self.EXTENT in parameters else "EXTENT"
+                )
+            )
 
         arguments = []
 
-        if not bbox.isNull():
+        # If it's a WMS layer and scale/DPI were given, craft a XML description file
+        if inLayer.providerType() == "wms" and isinstance(
+            parameters[self.INPUT], QgsProcessingRasterLayerDefinition
+        ):
+            # If the scale is greater than 0, we'll have a QgsProcessingRasterLayerDefinition
+            param_def = parameters[self.INPUT]
+            scale = param_def.referenceScale
+            dpi = param_def.dpi
+
+            distanceArea = None
+            if inLayer.crs().isGeographic():
+                distanceArea = QgsDistanceArea()
+                distanceArea.setSourceCrs(inLayer.crs(), context.transformContext())
+                distanceArea.setEllipsoid(inLayer.crs().ellipsoidAcronym())
+
+            width, height = GdalUtils._wms_dimensions_for_scale(
+                bbox, inLayer.crs(), scale, dpi, distanceArea
+            )
+            wms_description_file_path = tempfile.mktemp("_wms_description_file.xml")
```

Comment:
please use the processing utils api to ensure that the temporary file is placed in the correct location